public/video-ui/src/slices/workflow.ts

```diff
@@ -1,85 +1,118 @@
-              return void dispatch(statusNotFound(defaultWorkflowStatusData()));
-            }
-          });
-        } catch (e) {
-          // failed to parse response as json
-          return void dispatch(errorReceivingStatus(error));
-        }
+export const getStatus = createAsyncThunk<
+  FlatStub<string, string> | undefined,
+  { id: string }
+>('workflow/getStatus', (video, { dispatch }) =>
+  WorkflowApi.getAtomInWorkflow(video).catch((error: Response | any) => {
+    if (error instanceof Response) {
+      if (error.status !== 404) {
+        return void dispatch(
+          showError('Cannot get Atom status in Workflow', error)
+        );
+      }
+      try {
+        error.json().then(errorBody => {
+          if (
+            errorBody.errors &&
+            errorBody.errors.message === 'ContentNotFound'
+          ) {
+            return void dispatch(statusNotFound(defaultWorkflowStatusData()));
+          }
+        });
+      } catch (e) {
+        // failed to parse response as json
+        return void dispatch(
+          showError('Cannot get Atom status in Workflow', error)
+        );
       }
     }
-  )
+  })
 );
 
 export const getStatuses = createAsyncThunk<ExpandedStatus[] | undefined>(
   'workflow/getStatuses',
-  (_, { dispatch }) => WorkflowApi.getStatuses().catch(err => void dispatch(errorReceivingStatuses(err)))
+  (_, { dispatch }) =>
+    WorkflowApi.getStatuses().catch(
+      err =>
+        void dispatch(
+          showError(
+            `Could not get Workflow statuses. <a href="${WorkflowApi.workflowUrl}" target="_blank" rel="noopener">Open Workflow to get a cookie.</a>`,
+            err
+          )
+        )
+    )
 );
 
 export const getPriorities = createAsyncThunk<Priority[] | undefined>(
   'workflow/getPriorities',
-  (_, { dispatch }) => WorkflowApi.getPriorities().catch(err => void dispatch(errorReceivingPriorities(err)))
+  (_, { dispatch }) =>
+    WorkflowApi.getPriorities().catch(
+      err =>
+        void dispatch(
+          showError(
+            `Could not get Workflow priorities. <a href="${WorkflowApi.workflowUrl}" target="_blank" rel="noopener">Open Workflow to get a cookie.</a>`,
+            err
+          )
+        )
+    )
 );
 
-function errorReceivingPriorities(error: unknown) {
-  return showError(
-    `Could not get Workflow priorities. <a href="${WorkflowApi.workflowUrl}" target="_blank" rel="noopener">Open Workflow to get a cookie.</a>`,
-    error
-  );
-}
-
-
-function errorReceivingStatus(error: unknown) {
-  return showError(
-    'Cannot get Atom status in Workflow',
-    error
-  );
-}
+export const trackInWorkflow = createAsyncThunk<unknown, WorkflowDetails>(
+  'workflow/trackInWorkflow',
+  (workflowDetails, { dispatch }) =>
+    WorkflowApi.trackInWorkflow(workflowDetails).catch(err => {
+        dispatch(showError('Failed to track Atom in Workflow', err));
+        throw err;
```

Comment:
Throwing an error here means we can prevent the editor from transitioning to the readonly state.